gobblin-metastore/src/main/java/gobblin/metastore/FsStateStore.java

```diff
@@ -279,9 +276,13 @@ public void createAlias(String storeName, String original, String alias) throws
     }
 
     Path aliasTablePath = new Path(new Path(this.storeRootDir, storeName), alias);
+    Path tmpAliasTablePath = new Path(aliasTablePath.getParent(), new Path(TMP_FILE_PREFIX, aliasTablePath.getName()));
     // Make a copy of the original table as a work-around because
     // Hadoop version 1.2.1 has no support for symlink yet.
-    FileUtil.copy(this.fs, originalTablePath, this.fs, aliasTablePath, false, true, this.conf);
+    HadoopUtils.deleteIfExists(this.fs, tmpAliasTablePath, true);
+    HadoopUtils.copyPath(this.fs, originalTablePath, this.fs, tmpAliasTablePath, this.conf);
+    HadoopUtils.deleteIfExists(this.fs, aliasTablePath, true);
```

Comment:
If this line succeeds and the next line fails the state will be lost. One solution is to modify `FsDatasetStateStore.getLatestDatasetStatesByUrns` such that if `current.jst` doesn't exist, it will look for the most recent jst file.